### DISCUSSION TITLE
Field names on readContract() response

### DISCUSSION THREAD

**Discussion by R-is-G** [wevm/wagmi#2414] [Ideas]
It would be awesome if the response of the function readContract() could come with the names of the fields, like when you read contracts with web3js or ethers.js. I'll try to explain myself.

Using readContract() wagmi function, returns data like this:
![image](https://github.com/wagmi-dev/wagmi/assets/133210715/675250f6-66ee-4d79-b464-79115108702b)

And reading contract function with ethers (web3js is similar), returns data like this:
![image](https://github.com/wagmi-dev/wagmi/assets/133210715/2b3b572c-5285-4ff3-b873-db5bb64a6078)

To work with the result of the contract reading, it's much easier with the ethers/web3js response. I would be awesome if wagmi response could be with the field names.

**Comment by jxom** [comment]
There is a reason why it is like this! See https://github.com/wagmi-dev/viem/blob/main/.github/FAQ.md#why-is-a-contract-function-return-type-returning-an-array-instead-of-an-object

TLDR; we want to avoid ambiguous tuple types & "hacking" JavaScript to create an **unserializable** Array/Object type (this doesn't play well with SSR & local storage).

**Reply by R-is-G** [reply]
Thank you very much for the quick reply! With this I can solve my problem.
